src/dune_rules/install_rules.ml

```diff
@@ -1033,15 +1083,20 @@ include (
       ;;
 
       let action (entries, dst) ~ectx:_ ~eenv:_ =
-        let entries =
-          List.concat_map entries ~f:(fun (entry : _ Install.Entry.t) ->
-            match entry.kind with
-            | `File -> [ entry ]
-            | `Directory -> read_dir_recursively entry)
-          |> Install.Entry.gen_install_file
+        let open Fiber.O in
+        let+ entries =
+          let+ entries =
+            Fiber.parallel_map entries ~f:(fun (entry : _ Install.Entry.t) ->
+              match entry.kind with
+              | `File -> Fiber.return [ entry ]
+              | `Directory -> Fiber.return (read_dir_recursively entry)
+              | `Source_tree ->
+                (* these are expanded into [`File] *)
```

Comment:
nitpick: the `{` should be a `[`

Reply:
Did you attach your comment to the wrong line? There's no `{` on this line`.

Reply:
ah you're right, for some reason I saw the `[` as a `{`